Objects/TI4_Helpers/TI4_UnitHelper.ttslua

```diff
@@ -1008,34 +1041,44 @@ local _unitModifiers = {
         type = TYPE.CHOOSE,
         toggleActive = true,
         apply = function(unitAttrs, params)
+            local best = false
             for unitType, attr in pairs(unitAttrs) do
-                local best = false
-                if attr.antiFighterBarrage and (params.myUnitTypeToCount[unitType] or 0) > 0 then
+                -- Check for units present, then check for dynamically injected units, then default to 0
+                local unitCount = params.myUnitTypeToCount[unitType] or attr.unitCount or 0
+                if attr.antiFighterBarrage and unitCount > 0 then
                     if (not best) or (attr.antiFighterBarrage.hit < best.antiFighterBarrage.hit) then
                         best = attr
                     end
                 end
-                if best then
```

Comment:
NOTE: Moved the `if best then [...]` check outside of the for loop, where it should be.